pygfx/__init__.py

```diff
@@ -16,7 +16,7 @@
 from .renderers import *
 
 from .utils.color import Color
-from .utils.load import load_mesh, load_meshes, load_scene
+from .utils.load import load_mesh, load_meshes, load_scene, meshes_from_open3d
```

Comment:
To make the API somewhat consistent with regards to the functions privided by `pygfx.utils.load`, I think it makes sense if this'd be `load_meshed_from_open3d(path)` instead.

The geometry and material compat functions are fine, but the public function in `load.py` (the ones we import here, anyway) all work with a path. Does that make sense?

Reply:
`meshes_from_open3d` does not take a`Path` as argument, but a `TriangleMeshModel`, which is a combination of materials and meshes. So it's neither a geometry helper nor a material helper.

I can understand your concern and it was not clear to me where to put the method. I could create a `load_meshed_from_open3d(path)` which is using the current method, but I don't really see the benefit (other than having one more method). Trimesh is already doing a good job in `load_meshes`. The method `meshes_from_open3d(trianglemeshmodel)` is used to convert an already in-memory `TriangleMeshModel` into a `List[gfx.Mesh]`.

But I may not have the full picture here. My intention is to make it possible to use Open3D to load and process 3D models, because it's really good at that. But when it comes to visualisation, Open3D can be a real pain if you want to customise the rendering or the UI. That's where pygfx does a much better job, with a clear API and a lot of freedom. What I did not intend to do was to "replace" trimesh or gltflib completely.

Of course, I am open to other ideas and I could extend the integration to cover more cases (like loading from file and so on).

Reply:
> My intention is to make it possible to use Open3D to load and process 3D models, [...]

Right, and then you need something that takes the in-memory model, not a filename. Gotcha.

The thing I'm a bit worried about is that our API w.r.t. these load/conversion methods feels a bit inconsistent. There are multiple libs that can read meshes, and if we want to play nice with more of these, in a way that does not get messy, we need a better plan. Something to look into in the future ...

I think that for now, we can cancel this little change `from .utils.load import ... meshes_from_open3d`. This means that in the examples you'd have to do `gfx.load.meshes_from_open3d`. This is a bit less pretty, but when the time comes that we implement a better load/convert API, we can probably leave `meshes_from_open3d` intact, while if it would be in the `gfx` namespace, may want to deprecate it. 

In other words, I care a lot about keeping the `gfx` namespace clean, but I don't mind a lot if `gfx.load` gets multiple somewhat inconsistent functions 😄

Reply:
Maybe @Korijn has an opinion about this too.

Reply:
Yes, let's remove it from the global import to keep the load API consistent. I've already changed the code.
Is there any idea of moving the IO part for gfx into it's own library in the future?

Reply:
The important parts are:

* Ensure open3d imports are optional
* The compatibility layer between open3d objects and pygfx objects is what matters most
* Direct loading functions are for convenience
* Consistency between different libraries is a pipe dream

Reply:
> Is there any idea of moving the IO part for gfx into it's own library in the future?

I don't think there's much use. We might as well pick one of these libs and award it with "the official pygfx mesh io lib" 😉. And as Korijn said, abstracting over all io libraries is not feasible.

Reply:
I made an issue to discuss how to move forward with io: https://github.com/pygfx/pygfx/issues/888

Reply:
Sounds good to me. I think the PR now fulfils the requirements.